packages/babel-plugin-transform-es2015-modules-umd/src/index.js

```diff
@@ -65,23 +74,74 @@ export default function ({ types: t }) {
             } else if (arg.value === "exports") {
               return t.memberExpression(t.identifier("mod"), t.identifier("exports"));
             } else {
-              let requireName = basename(arg.value, extname(arg.value));
-              let globalName = browserGlobals[requireName] || requireName;
+              let memberExpression;
 
-              return t.memberExpression(t.identifier("global"), t.identifier(
-                t.toIdentifier(globalName)
-              ));
+              if (state.opts.exactGlobals) {
+                let globalRef = browserGlobals[arg.value];
+                if (globalRef) {
+                  if (globalRef.indexOf(".") > -1) {
+                    memberExpression = globalRef.split(".").reduce(
+                      (accum, curr) => t.memberExpression(accum, t.identifier(curr)), t.identifier("global")
+                    );
+                  } else {
+                    memberExpression = t.memberExpression(t.identifier("global"), t.identifier(globalRef));
+                  }
+                } else {
+                  memberExpression = t.memberExpression(
+                    t.identifier("global"), t.identifier(t.toIdentifier(arg.value))
+                  );
+                }
+              } else {
+                let requireName = basename(arg.value, extname(arg.value));
+                let globalName = browserGlobals[requireName] || requireName;
+                memberExpression = t.memberExpression(
+                  t.identifier("global"), t.identifier(t.toIdentifier(globalName))
+                );
+              }
+
+              return memberExpression;
             }
           });
 
-          let globalArg = t.identifier(t.toIdentifier(moduleName ? moduleName.value : this.file.opts.basename));
+          let moduleNameOrBasename = moduleName ? moduleName.value : this.file.opts.basename;
+          let globalToAssign = t.memberExpression(
+            t.identifier("global"), t.identifier(t.toIdentifier(moduleNameOrBasename))
+          );
+          let prerequisiteAssignments = null;
+
+          if (state.opts.exactGlobals) {
+            let globalName = browserGlobals[moduleNameOrBasename];
+
+            if (globalName) {
+              if (globalName.indexOf(".") > -1) {
```

Comment:
Same here since if there is just one `.`, it'll `.reduce` over an empty array and return

    t.memberExpression(t.identifier("global"), t.identifier(members[0]))

which is just what your `else` is doing.